test/compress/reduce_vars.js

```diff
@@ -3379,3 +3379,75 @@ issue_2420_2: {
         "false false true",
     ]
 }
+
+issue_2423_1: {
+    options = {
+        reduce_vars: true,
+        toplevel: true,
+        unused: true,
+    }
+    input: {
+        function c() { return 1; }
+        function p() { console.log(c()); }
+        p();
+        p();
+    }
+    expect: {
+        function p() { console.log(function() { return 1; }()); }
+        p();
+        p();
+    }
```

Comment:
In a followup commit could you please add `expect_stdout: true` to these tests?

Reply:
Done in #2430